fmutils.go

```diff
@@ -170,30 +170,57 @@ func (mask NestedMask) Overwrite(src, dest proto.Message) {
 	mask.overwrite(src.ProtoReflect(), dest.ProtoReflect())
 }
 
-func (mask NestedMask) overwrite(src, dest protoreflect.Message) {
-	for k, v := range mask {
-		srcFD := src.Descriptor().Fields().ByName(protoreflect.Name(k))
-		destFD := dest.Descriptor().Fields().ByName(protoreflect.Name(k))
-		if srcFD == nil || destFD == nil {
-			continue
-		}
-
-		// Leaf mask -> copy value from src to dest
-		if len(v) == 0 {
-			if srcFD.Kind() == destFD.Kind() { // TODO: Full type equality check
-				val := src.Get(srcFD)
-				if isValid(srcFD, val) {
-					dest.Set(destFD, val)
+func (mask NestedMask) overwrite(srcRft, destRft protoreflect.Message) {
+	for srcFDName, submask := range mask {
+		srcFD := srcRft.Descriptor().Fields().ByName(protoreflect.Name(srcFDName))
+		srcVal := srcRft.Get(srcFD)
+		if len(submask) == 0 {
+			if isValid(srcFD, srcVal) {
+				destRft.Set(srcFD, srcVal)
+			} else {
+				destRft.Clear(srcFD)
+			}
+		} else if srcFD.IsMap() && srcFD.Kind() == protoreflect.MessageKind {
+			srcMap := srcRft.Get(srcFD).Map()
```

Comment:
Please add a test for this case (when a source field is a map).

See the coverage report: https://app.codecov.io/gh/mennanov/fmutils/pull/10?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=Renat

Reply:
Fixed to have 100% test case coverage.